corehq/apps/ota/views.py

```diff
@@ -106,6 +107,8 @@ def search(request, domain):
 
     try:
         hits = search_es.run().raw_hits
+    except TooManyRelatedCasesException:
+        return HttpResponse(_('Search has too many results. Please try a more specific search.'), status=400)
```

Comment:
Yeah, I had trouble coming up with something that would be accurate but also useful to end users, who aren't going to be familiar with anything like "parent". I could try using the display text of the property, like "Please try a more specific value for MOH id." It's also possible that it's just going to be the app builder's responsibility to not configure parent fields that could match a lot of cases. Open to suggestions, and I can ping the product team about it.

Reply:
I realized this code doesn't have access to the field display text, since it doesn't know which module is issuing the search. I could include the field path, but an error referencing something like `parent/moh_id` is rather wonky. Leaving this alone for now.